### PR TITLE
MINIFI-492 - Generic reporting tasks

### PR DESCRIPTION
**Pull request by nghiaxlee**
Thank you for submitting a contribution to Apache NiFi - MiNiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi-minifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under minifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under minifi-assembly?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java [hunk 1]
```diff
@@ -508,34 +500,29 @@ protected static void addFunnel(final Element parentElement, FunnelSchema funnel
...
+            addTextElement(element, "id", reportingSchema.getId());
+            addTextElement(element, "name", reportingSchema.getName());
+            addTextElement(element, "comment", reportingSchema.getComment());
+            addTextElement(element, "class", reportingSchema.getReportingClass());
+            addTextElement(element, "schedulingPeriod", reportingSchema.getSchedulingPeriod());
+            addTextElement(element, "scheduledState", "RUNNING");
+            addTextElement(element, "schedulingStrategy", reportingSchema.getSchedulingStrategy());
 
-            addConfiguration(taskElement, attributes);
+            // TODO: Clarify this
```
Comment (apiri) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java:517]: given the generic (and correct) handling of the properties below, I believe we need to see if we are performing this override and provide special handling.

File: minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ReportingSchema.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.nifi.scheduling.SchedulingStrategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.apache.nifi.minifi.commons.schema.common.CommonPropertyKeys.*;
+import static org.apache.nifi.minifi.commons.schema.ProcessorSchema.IT_IS_NOT_A_VALID_SCHEDULING_STRATEGY;
+
+/**
```
Comment (apiri) [inline minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ReportingSchema.java:30]: Would give a description here or remove

File: minifi-toolkit/minifi-toolkit-configuration/src/test/resources/1.5_RPG_Handling.yml [hunk 3]
```diff
@@ -112,3 +112,5 @@ Remote Process Groups:
     use compression: false
   Output Ports: []
 NiFi Properties Overrides: {}
+Reporting Tasks: []
```
Comment (apiri) [inline minifi-toolkit/minifi-toolkit-configuration/src/test/resources/1.5_RPG_Handling.yml:115]: Seems this was duplicated.  I also think it should be okay to not have a Reporting Tasks section from a config standpoint.  For existing flows, we do not wish to break them.

Comment (nghiaxlee) [reply minifi-toolkit/minifi-toolkit-configuration/src/test/resources/1.5_RPG_Handling.yml]: Agree, I reverted the changes

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java (hunk 1 above)
Comment (nghiaxlee) [reply minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java]: For backward compatibility, if SSL is enabled, the Provenance Reporting task declared in `Provenance Reporting` node will have secure connect as well. If a provenance reporting task is declared in `Reporting Tasks:` node, user has to set it manually.

**Comment by apiri** [comment]
Updates look good.  +1

**Pull request closed without merging**